tutorials/plot_background_statistics.py

```diff
@@ -0,0 +1,642 @@
+#    ``threshold`` value.
+# 2. Cluster points that exceed this threshold (with the same sign)
+#    based on adjacency.
+# 3. Record the *size* of each cluster (measured, e.g., by a simple voxel
+#    count, or by the sum of voxel t-values within the cluster).
+#
+# After doing these permutations, the cluster sizes in our veridical data
+# are compared to this null distribution. The p-value associated with each
+# cluster is again given by the proportion of smaller null distribution
+# values. This can then be subjected to a standard p-value threshold
+# (e.g., ``p < 0.05``) to reject the null hypothesis (i.e., find an effect
+# of interest).
+#
+# This reframing to consider *cluster sizes* rather than *individual means*
+# maintains the advantages of the standard non-parametric permutation
+# test -- namely controlling FWER and making no assumptions of parametric
+# data distribution.
+# Cricitally, though, it also accounts for the correlation structure in the
+# data -- which in this toy case is spatial but
+# in general can be multidimensional (e.g., spatio-temporal) -- because the
+# null distribution will be derived from data in a way that preserves these
+# correlations.
+#
+# However, there is a drawback. If a cluster significantly deviates from
+# the null, no further inference on the cluster (e.g., peak location) can be
+# made, as the entire cluster as a whole is used to reject the null.
+# Moreover, because the test statistic concerns the full data, the null
+# hypothesis (and our rejection of it) refers to the structure of the full
+# data. For more information, see also the
+# `excellent FieldTrip cluster interpretation tutorial <ft_cluster>`_.
+#
+# Defining the connectivity/neighbor/adjacency matrix
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# First we need to define our connectivity/neighbor/adjacency matrix.
+# This is a square array (or sparse matrix) of shape ``(n_src, n_src)`` that
+# contains zeros and ones to define which spatial points. In our case this
+# is quite simple, as our data are aligned on a grid.
+#
+# Let's pretend that our data were on a 3x3 grid. Thinking about each voxel
+# as being connected to the other voxels it touches, we would need a 9x9
+# connectivity matrix. The first row should contain the elements in the
+# ``.ravel()``'ed data that it touches. Since it touches the second element
+# in the first row and the first element in the second row (and is also
+# a neighbor to itself), this would be::
+#
+#     [1, 1, 0, 1, 0, 0, 0, 0, 0]
+#
+# :mod:`sklearn.feature_extraction` provides a convenient function for this:
+
+from sklearn.feature_extraction.image import grid_to_graph  # noqa: E402
+mini_connectivity = grid_to_graph(3, 3).toarray()
+assert mini_connectivity.shape == (9, 9)
+print(mini_connectivity[0])
+del mini_connectivity
+
+###############################################################################
+# In general the connectivity betwene voxels can be more complex.
```

Comment:
between

Reply:
Actually, how much does the `del` help here? Memory usage is presumably minimal, so the line might be superfluous clutter.

Reply:
it's not about memory, it's cleaning the namespace to make sure people don't think it should actually be used